pkg/apis/k3k.io/v1alpha1/types.go

```diff
@@ -14,21 +14,26 @@ type Cluster struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 	metav1.TypeMeta   `json:",inline"`
 
+	// +kubebuilder:default={}
 	Spec   ClusterSpec   `json:"spec"`
 	Status ClusterStatus `json:"status,omitempty"`
 }
 
 type ClusterSpec struct {
 	// Version is a string representing the Kubernetes version to be used by the virtual nodes.
-	Version string `json:"version"`
+	Version string `json:"version,omitempty"`
```

Comment:
What happens if you don't set the `Version`?

It looks like https://github.com/rancher/k3k/blob/e7671134d26c94e950bde553e307702d0ad90ffd/pkg/controller/controller.go#L30 would generate an invalid image reference?

So, this should probably be `// +required` and **not** `omitempty` ?

Reply:
@bigkevmcd that's correct. The idea was to simplify all the possible fields, and the empty version will fallback to the same host k3s version. To handle this the controller will check this, update, and requeue the resource.

https://github.com/rancher/k3k/blob/a00c9e243a0e35e8a0b0bec6ee9c6379d9379561/pkg/controller/cluster/cluster.go#L90-L102

I'm not completely sold on this, so I'm happy to discuss any possible issue with that.

On the UX perspective this is something that can be handled via the UI, or with the `k3kcli`, keeping the version as the only required field.

Reply:
Feels like it's better to prevent it being created/updated, than to fail to reconcile?

Reply:
I think what I'd do here, is leave the spec.version empty, but populate the status with the version.